main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -509,6 +510,43 @@ object CompletionProvider {
     }
   }
 
+  /** 
+    * Returns a list of completion items based on case keyword in match expressions
+    */
+  private def getCaseCompletions()(implicit context: Context, index: Index, root: TypedAst.Root): Iterable[CompletionItem] = {
+    if (root == null) {
+      return Nil
+    }
+    
+    //TODO Do not match when declaring enums.
+    val casePattern = raw"\s*ca?s?e?\s?.*".r
+    val wordPattern = "ca?s?e?".r
+
+    val currentWordIsCase = wordPattern matches context.word
+
+    //Currently only suggests enum cases inside the same source file.
+    val iter = index.query(context.uri)
+
+    if (casePattern matches context.prefix) {
+      for {
+        (sym, _) <- iter
+          .flatMap(entity => entity match {
+            case Entity.Enum(e) => e.cases
+            case _ => Map()
+          })
+        name = sym.name
+        completion = if (currentWordIsCase) s"case $name => " else s"$name => "
```

Comment:
I think it should be possible to complete more than just the tag name, but also its content. E.g. if you have:

```
enum Shape {
  case Rectangle(Int32, Int32)
```

then we want to complete more than just Rectangle.